aws/cloudformation/cloud_formation_stack.yml.erb

```diff
@@ -0,0 +1,215 @@
+<%
+require 'cdo/aws/cloudfront'
+require 'aws-sdk'
+%>
+---
+AWSTemplateFormatVersion: 2010-09-09
+Description: AWS CloudFormation Template for Code.org application
+Resources:
+  VPC:
+    Type: AWS::EC2::VPC
```

Comment:
Yeah, I'm still trying to figure out the best approach to manage divergent and increasingly-complex templates long-term. For now, I thought the full-production stack would be significantly different from the adhoc-standalone stack, which is why I built them up separately to start. We will definitely need to find a way to factor out the duplicated parts eventually, though, and there are two options I'm considering currently.

First, AWS has some [Best Practices](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/best-practices.html) recommendations which may or may not be helpful. Specifically, they recommend using [nested stacks](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/best-practices.html#nested) as a way to separate out reusable components. However, this adds a layer of complexity in itself (the inputs and outputs of the nested stack need to be explicitly defined and can become quite verbose), plus it adds an extra layer of CloudFormation API calls, event-reporting entries, and Stack resources in the list view.

An alternative would be to introduce a simple `partial` function in the ERB filter similar to what we use in Pegasus-Sinatra templates (see [`text_render.rb`](https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/pegasus/text_render.rb#L34-L49)) to separate out reusable components. This adds a bit of ruby complexity to the enhanced template system however, so there's a bit of a tradeoff.